msdk-rawdata/msdk-rawdata-centroiding/src/main/java/io/github/msdk/rawdata/centroiding/WaveletCentroidingAlgorithm.java

```diff
@@ -90,14 +81,12 @@ public WaveletCentroidingAlgorithm(@Nonnull Integer scaleLevel, @Nonnull Double
   }
 
   /**
-   * Perform the CWT over raw data points in the selected scale level
-   * 
-   * @param dataPoints
+   * Perform the CWT over raw intensities in the selected scale level
    */
-  private void performCWT() {
+  private static @Nonnull float[] performCWT(@Nonnull float[] intensityBuffer, int numOfDataPoints,
+      int scaleLevel, double waveletWindow) {
 
-    if (cwtDataPoints.length < numOfDataPoints)
```

Comment:
BTW, this line originally threw a NPE because the array wasn't initialized. Just as FindBugs said would be a problem.